pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -67,17 +67,67 @@ type topologyPairsMaps struct {
 	podToTopologyPairs map[string]topologyPairSet
 }
 
-// topologyPairsPodSpreadMap combines topologyKeyToMinPodsMap and topologyPairsMaps
+type criticalPath struct {
+	// topologyValue denotes the topology value mapping to topology key.
+	topologyValue string
+	// matchNum denotes the number of matching pods.
+	matchNum int32
+}
+
+// CAVEAT: the reason that `[2]criticalPath` can work is based on the implementation of current
+// preemption algorithm - we only preempt pods on the same node, instead of pods on multiple nodes.
+// If we plan to turn to a more complex algorithm like "arbitrary pods on multiple nodes", this
+// structure needs to be revisited.
```

Comment:
we need to also clarify that this works also because each node is evaluated on a separate copy of the metadata.

Reply:
SG.